### DISCUSSION TITLE
Region dependent key bindings

### DISCUSSION THREAD

**Discussion by witek** [meow-edit/meow#424] [Ideas]
I have bound the key `w` to `meow-mark-symbol`. Now when I press `w` a symbol is marked. When i press `w` again nothing changes. It would be nice if `w` would `meow-next-symbol` when a region already exists.

Is there a simple way to set this up?

Is there a reason not to have such region dependent key bindings?

**Comment by witek** [comment]
> Is there a simple way to set this up?


```lisp
(defun my-meow-mark-symbol ()
  (interactive)
  (if (use-region-p)
      (call-interactively 'meow-next-symbol)
    (call-interactively 'meow-mark-symbol)))
```

**Comment by DogLooksGood** [comment]
Well, you can have any command you like, there's really no restriction here.  Personally, I'm against "Do-What-I-Mean" style command.  With a DWIM style command, you call once it does one thing, you call again, it does something else.  I want every commands work as how they are described.  If we have a command that named as "mark a symbol", it should mark a symbol, that's it. 

You may notice there are some "DWIM" behaviors in Meow, for example, you can use `meow-kill` to kill a region, without a region, it kills to the end of line.  That's because we have a mechanism for no-region fallback behavior, it's decoupled and configurable.

**Comment by sacereda** [comment]
Hi,
Is there anything similar to this behaviour in objed.el? 
https://github.com/clemera/objed#examples-of-usage

This seems useful in order to move from one thing to next/prev. I've found myself implementing this in normal mode:

   '("[" . backward-paragraph)
   '("]" . forward-paragraph)

But maybe something like this could work?

- meow-inner-of-thing
- paragraph
- meow-next-thing

**Reply by DogLooksGood** [reply]
No, there isn't.   Just use built-in commands for paragraphs, sentences movement. 
Check this: https://github.com/meow-edit/meow/issues/127#issuecomment-992777150

**Comment by sacereda** [comment]
Oh, I didn't check that one. Thanks!
